libdrgn/program.c

```diff
@@ -995,3 +995,86 @@ drgn_program_member_info(struct drgn_program *prog, struct drgn_type *type,
 	ret->bit_field_size = member->bit_field_size;
 	return NULL;
 }
+
+struct drgn_string_symbol {
+	const char *name;
+	struct drgn_symbol *symbol;
+	struct drgn_error *err;
+};
+
+static int drgn_dwarf_module_iterate_sym_table(
+	Dwfl_Module *dwfl_module,
+	void **userdatap,
+	const char *module_name,
+	Dwarf_Addr base, void *arg)
+{
+	GElf_Sym elf_sym;
+	GElf_Addr elf_addr;
+	struct drgn_string_symbol *sym;
+	int symtab_offset, symtab_len;
+
+	sym = arg;
+	sym->err = NULL;
+
+	symtab_len = dwfl_module_getsymtab(dwfl_module);
+	symtab_offset = dwfl_module_getsymtab_first_global(dwfl_module);
+
+	if (symtab_len == -1 || symtab_offset == -1) {
+		sym->err = drgn_error_format(DRGN_ERROR_LOOKUP,
+			"could not look up symbol table for %s.", module_name);
```

Comment:
If there are multiple modules with bad symbol tables, this leaks the previous error. I think it would suffice to have a flag that indicates that some symbol tables couldn't be read.